state/options/tx_options.go

```diff
@@ -0,0 +1,201 @@
+package options
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/celestiaorg/celestia-app/pkg/user"
+	apptypes "github.com/celestiaorg/celestia-app/x/blob/types"
+)
+
+const (
+	// gasMultiplier is used to increase gas limit in case if tx has additional options.
+	gasMultiplier = 1.1
+
+	//	Since 0 is a valid fee input for the Tx, the default value is -1.
+	defaultFeeAmount = -1
+)
+
+var (
+	errNoGasProvided     = errors.New("gas limit was not set")
+	errNoAddressProvided = errors.New("address was not set")
+)
+
+// TxOptions specifies additional options that will be applied to the Tx.
+type TxOptions struct {
+	// fee is private since it has to be set through `SetFeeAmount`
+	fee      *Fee
```

Comment:
Perhaps now would be a good time to advocate that the user should specify the gasPrice and not the total fee since it is the gas price that determines priority and inclusion.

Reply:
This is something we discussed in DM last week. It makes a lot of sense for me but since the implementation was done, I decided to push all changes w/o `minGasPrice` and discuss it here.
@distractedm1nd , @renaynay wdyt?

also @jcstein for visibility.

Reply:
Either we only have fee (without GasLimit), or we have GasPrice+Gas. I definitely prefer GasPrice+Gas